src/components/ImportStatus.js

```diff
@@ -0,0 +1,70 @@
+import _ from 'lodash/fp'
+import { Fragment } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import { clearNotification, notify } from 'src/components/Notifications'
+import { Ajax, useCancellation } from 'src/libs/ajax'
+import { withErrorReporting } from 'src/libs/error'
+import { pfbImportJobStore } from 'src/libs/state'
+import { DEFAULT, delay, switchCase, useAtom, useOnMount } from 'src/libs/utils'
+
+
+const ImportStatus = () => {
+  const jobs = useAtom(pfbImportJobStore)
+  return h(Fragment, _.map(job => h(ImportStatusItem, {
+    job,
+    onDone: () => {
+      pfbImportJobStore.update(_.reject({ jobId: job.jobId }))
+    }
+  }), jobs))
+}
+
+const ImportStatusItem = ({ job, onDone }) => {
+  const { targetWorkspace, jobId } = job
+  const signal = useCancellation()
+
+  useOnMount(() => {
+    const poll = withErrorReporting('Problem checking status of PFB data import', async () => {
+      while (true) {
+        await delay(5000)
+        await checkForCompletion(targetWorkspace, jobId)
+      }
+    })
+    poll()
+  })
+
+  const checkForCompletion = async ({ namespace, name }, jobId) => {
+    const { Workspaces } = Ajax(signal)
+
+    const fetchImportStatus = async () => {
+      try {
+        return await Workspaces.workspace(namespace, name).importPFBStatus(jobId)
+      } catch (error) {
+        // Ignore 404; We're probably asking for status before the status endpoint knows about the job
+        if (error.status === 404) {
+          return { status: 'PENDING' }
+        } else {
+          onDone()
+          throw error
+        }
+      }
+    }
+
+    const response = await fetchImportStatus()
+    const { message, status } = response
```

Comment:
Combine destructure

Reply:
Response is used below in the `DEFAULT` case.